store.go

```diff
@@ -1421,20 +1444,8 @@ func (s *store) canUseShifting(uidmap, gidmap []idtools.IDMap) bool {
 	return true
 }
 
-func (s *store) PutLayer(id, parent string, names []string, mountLabel string, writeable bool, lOptions *LayerOptions, diff io.Reader) (*Layer, int64, error) {
+func (s *store) putLayerLocked(rlstore rwLayerStore, rlstores []roLayerStore, id, parent string, names []string, mountLabel string, writeable bool, lOptions *LayerOptions, diff io.Reader, slo *stagedLayerOptions) (*Layer, int64, error) {
```

Comment:
Absolutely non-blocking: This could be `putLayer(… diff, slo)` which also does the locking; then `PutLayer` and `ApplyStagedLayer` would be very trivial wrappers. This works fine as it is.

Reply:
that is better, amended in the last version